src/meta/src/stream/stream_manager.rs

```diff
@@ -288,26 +292,45 @@ where
             .await?;
 
         let mut env = Env {
-            upstream_parallel_unit_info,
+            upstream_fragment_vnode_info,
+            upstream_vnode_bitmap_info,
             tables_worker_actors,
             locations,
             dispatchers,
+            actor_vnode_bitmaps: &mut Default::default(),
             upstream_worker_actors,
         };
 
         for fragment in table_fragments.fragments.values_mut() {
+            if !fragment_upstream_table_map.contains_key(&fragment.fragment_id) {
+                continue;
+            }
```

Comment:
Seems like the map is only used for fragments containing `Chain`s. How about introducing `FragmentType::Chain`?

Reply:
I tried, but found that `FragmentType::Chain` and `FragmentType::Sink` might appear in the same fragment. 😓

Reply:
That's very interesting! 😄 We may need to introduce a series of flags in the future. 🥵